starboard/BUILD.gn

```diff
@@ -185,6 +184,8 @@ source_set("starboard_headers_only") {
 }
 
 if (current_toolchain == starboard_toolchain) {
+  import("//starboard/build/config/starboard_target_type.gni")
```

Comment:
Is there any reason to limit this import? I think in general it's nice to have them grouped at the top and not have them be this dynamic (I think of it like python imports)

Reply:
i see, i did this after i added an assert for building with starboard toolchain  starboard_target_type.gni - which feels like a  good change

Reply:
There's no need to require that the toolchain be the starboard toolchain for starboard_target_type.gni—it's not like it defines any targets that are only built by starboard (just a template), and I can see something in the default toolchain potentially needing the GN variables from that file at some point.